cmds/cmd_functions/functions.js

```diff
@@ -18,14 +24,14 @@ module.exports = (argv, clusterName) => {
     function alreadyRegisteredCheck(jobContents) {
         if (_.has(jobContents, 'tjm.cluster')) {
             return teraslice.jobs.wrap(jobContents.tjm.job_id).spec()
-                .then((jobSpec) => {    
+                .then((jobSpec) => {
                     if (jobSpec.job_id === jobContents.tjm.job_id) {
-                        return Promise.resolve(true);
+                        return Promise.resolve();
                     }
-                    return Promise.resolve(false);
+                    return Promise.reject(new Error('Job is not on the cluster'));
                 });
         }
-        return Promise.resolve(false);
+        return Promise.reject(new Error('No cluster configuration for this job'));
     }
```

Comment:
Actually now that I think of it, I don't think this should return an error if the job isn't on the cluster just in cases where the job is new then it would just be some different logic.  Makes it easier instead of having to run register for every new job, the jobs would be automatically registered and then continue.  This may only apply to run.